kernel/idle.c

```diff
@@ -91,6 +93,9 @@ void idle(void *unused1, void *unused2, void *unused3)
 		}
 # endif /* !defined(CONFIG_USE_SWITCH) || defined(CONFIG_SPARC) */
 #endif /* !defined(CONFIG_PREEMPT_ENABLED) */
+#else /* !CONFIG_IDLE_ROUTINE_DEFAULT */
+	idle_enter_hook();
+#endif /* CONFIG_IDLE_ROUTINE_DEFAULT */
```

Comment:
This change disconnects system power management. That is a showstopper for me. Basically when this idle hook is enabled pm is never kicked or has to be done manually by anyone implementing it. For me would be better if we have an enter hook that depending on the returned value does not starts the pm logic and optionally we have an exit hook for possible cleanup. Alternatively we could introduce soc_cpu_idle one layer down. Something like `soc_idle()` -> `cpu_idle()`. The former option is better IMHO.

Reply:
Disconnecting PM here is just a PoC that SoCs which don't have complex power states, like many nRF SoCs, can implement some simpler idle logic, while being compatible with the event device API. It will go away long before this PR is merged :)

Reply:
can we remove/change it then ? So we can move on this pr